src/dmd/dtemplate.d

```diff
@@ -519,6 +519,7 @@ extern (C++) final class TemplateDeclaration : ScopeDsymbol
     bool ismixin;           // template declaration is only to be used as a mixin
     bool isstatic;          // this is static template declaration
     Prot protection;
+    int inuse;              // for recursive expansion detection
```

Comment:
Please add a Ddoc comment and I suggest naming it `inUse` to follow standard D naming conventions.

Reply:
I veto renaming, as it doesn't follow the internal compiler convention, and conflicts with naming convention for functions - there should be a distinction between what is and is not a side effect, especially as not everyone is using parentheses anymore.

Reply:
I think it would improve the DMD code base to establish a style guide of some kind.  [The official style guide](https://dlang.org/dstyle.html) is a good place to start.  If it is insufficient in some way, let's amend it for DMD.  DMD's code has clearly gone through a few stages, and is showing its age.  It would be nice to set some code-quality expectations for new code, and encourage refactorings that would help remove some of the oddities.

Perhaps now is not the right time with the backend still written in C+.  but I hope we can come to some agreement about conventions and document them somewhere. 

 The point about side effects is interesting, but isn't `pure` supposed to help with that?

Reply:
I don't see how the backend being written in C++ affects the style guide of the D code.

Reply:
I wouldn't say that renaming is required but I do require the Ddoc comment. Or make it `private`, it's not used anywhere outside of the class as far as I can see.

Reply:
> The point about side effects is interesting, but isn't pure supposed to help with that?

Most methods in the compiler are pure, because they only change internal state.  (Pure in the D sense is not to be confused with pure in the C sense, or absent of side effects).